frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx

```diff
@@ -124,10 +162,58 @@ export const ClusterServiceVersionTableRow = withFallback<ClusterServiceVersionT
   );
 });
 
+export const FailedSubscriptionTableRow: React.FC<FailedSubscriptionTableRowProps> = (props) => {
+  const {obj, index, key, style} = props;
+
+  const route = resourceObjPath(obj, referenceForModel(SubscriptionModel));
+  const menuActions = [Kebab.factory.Edit, () => uninstall(obj)];
+  const subscriptionState = _.get(obj.status, 'state', 'Unknown');
+
+  return <TableRow id={obj.metadata.uid} index={index} trKey={key} style={style}>
+    <TableData className={tableColumnClasses[0]}>
+      <Link to={route}>
+        <ClusterServiceVersionLogo icon={null} displayName={obj.spec.name} version={null} provider={null} />
+      </Link>
+    </TableData>
+    <TableData className={tableColumnClasses[1]}>
+      <ResourceLink kind="Namespace" title={obj.metadata.namespace} name={obj.metadata.namespace} />
+    </TableData>
+    <TableData className={tableColumnClasses[2]}>
+      <span className="text-muted">None</span>
+    </TableData>
+    <TableData className={tableColumnClasses[3]}>
+      { ['Unknown', SubscriptionState.SubscriptionStateFailed].includes(subscriptionState) && <span className="co-icon-and-text co-error">
+        <ErrorStatus title={subscriptionState} />
+      </span> }
+      { subscriptionState === SubscriptionState.SubscriptionStateUpgradePending && <span className="co-icon-and-text">
+        <ProgressStatus title={subscriptionState} />
+      </span>}
+    </TableData>
+    <TableData className={tableColumnClasses[4]}>
+      <span className="text-muted">None</span>
+    </TableData>
+    <TableData className={tableColumnClasses[5]}>
+      <ResourceKebab resource={obj} kind={referenceFor(obj)} actions={menuActions} />
+    </TableData>
+  </TableRow>;
+};
+
+const subscriptionFor = (csv: ClusterServiceVersionKind) => (subs: SubscriptionKind[]) => subs.find(sub => {
+  return sub.metadata.namespace === csv.metadata.annotations['olm.operatorNamespace'] && _.get(sub.status, 'installedCSV') === csv.metadata.name;
+});
+
 export const ClusterServiceVersionList: React.SFC<ClusterServiceVersionListProps> = (props) => {
   const EmptyMsg = () => <MsgBox title="No Cluster Service Versions Found" detail="" />;
 
-  return <Table {...props} aria-label="Installed Operators" Header={ClusterServiceVersionTableHeader} Row={ClusterServiceVersionTableRow} EmptyMsg={EmptyMsg} virtualize />;
+  return <Table
+    {...props}
+    aria-label="Installed Operators"
+    Header={ClusterServiceVersionTableHeader}
+    Row={(rowProps) => referenceFor(rowProps.obj) === referenceForModel(ClusterServiceVersionModel)
```

Comment:
Use different `props.Row` component depending on if object is a `ClusterServiceVersion` or failed `Subscription`.